src/inmanta/agent/handler.py

```diff
@@ -137,9 +138,15 @@ def wrapper(self, *args, **kwds):
             def bound(**kwds):
                 return f(self, **kwds)
 
-            cache_none = cacheNone
             return self.cache.get_or_else(
-                f.__name__, bound, for_version, timeout, myignore, cache_none, **kwds, call_on_delete=call_on_delete
+                f.__name__,
+                bound,
+                for_version,
+                timeout,
+                myignore,
+                cacheNone if cacheNone is not None else cache_none,
```

Comment:
Interesting note: the suggested implementation apparently had scoping issues:
```
if cacheNone is not None:
    cache_none = cacheNone
```
makes `cache_none` a local variable because we're in a closure. `cache_none = cacheNone if cacheNone is not None else cache_none` doesn't work either for the same reasons. So I learned a bit more of the fine points of Python variable scoping.

Reply:
cool!